build-support/functions/20-build.sh

```diff
@@ -0,0 +1,402 @@
+function refresh_docker_images {
+   # Arguments:
+   #   $1 - Path to top level Consul source
+   #   $2 - Which make target to invoke (optional)
+   #
+   # Return:
+   #   0 - success
+   #   * - failure
+   
+   if ! test -d "$1"
+   then
+      err "ERROR: '$1' is not a directory. refresh_docker_images must be called with the path to the top level source as the first argument'" 
+      return 1
+   fi
+   
+   local sdir="$1"
+   local targets="$2"
+   
+   test -n "${targets}" || targets="docker-images"
+   
+   make -C "${sdir}" ${targets}
+   return $?
+}
+
+function build_ui {
+   # Arguments:
+   #   $1 - Path to the top level Consul source
+   #   $2 - The docker image to run the build within (optional)
+   #
+   # Returns:
+   #   0 - success
+   #   * - error
+   #
+   # Notes:
+   #   Use the GIT_COMMIT environment variable to pass off to the build
+   
+   if ! test -d "$1"
+   then
+      err "ERROR: '$1' is not a directory. build_ui must be called with the path to the top level source as the first argument'" 
+      return 1
+   fi
+   
+   local image_name=${UI_BUILD_CONTAINER_DEFAULT}
+   if test -n "$2"
+   then
+      image_name="$2"
+   fi   
+   
+   local sdir="$1"
+   local ui_dir="${1}/ui-v2"
+   
+   # parse the version
+   version=$(parse_version "${sdir}")
+   
+   local commit_hash="${GIT_COMMIT}"
+   if test -z "${commit_hash}"
+   then
+      commit_hash=$(git rev-parse --short HEAD)
+   fi
+   
+   # make sure we run within the ui dir
+   pushd ${ui_dir} > /dev/null
+   
+   status "Creating the UI Build Container with image: ${image_name}"
+   local container_id=$(docker create -it -e "CONSUL_GIT_SHA=${commit_hash}" -e "CONSUL_VERSION=${version}" ${image_name})
+   local ret=$?
+   if test $ret -eq 0
+   then
+      status "Copying the source from '${ui_dir}' to /consul-src within the container"
+      (
+         docker cp . ${container_id}:/consul-src &&
+         status "Running build in container" && docker start -i ${container_id} &&
+         rm -rf ${1}/ui-v2/dist &&
+         status "Copying back artifacts" && docker cp ${container_id}:/consul-src/dist ${1}/ui-v2/dist
```

Comment:
See comment below on the `mv`ing. Need to follow around what things are doing here properly, but if the dockerfile runs `make build` essentially in the ui-v2 folder, the 'mv' in the GNUMakefile below will try to move the produced `dist` to `/pkg/web_ui/v2`, maybe I've following it wrong, ah I suppose that line fails now as `/pkg/web_ui/` doesn't exist in the container? Pretty sure we can just remove the `mv` line from the /ui-v2/GNUMakefile below?

Reply:
Ah, GH lists things by comment time, exchange 'comment below' for 'comment above' 😄 ^

Reply:
Nothing is calling make dist in there any longer. We can probably just remove it all together.